From: Yuval Shaia <yuval.shaia@oracle.com>
Date: Fri, 21 Jul 2017 22:20:50 +0300
Subject: IB/rxe: Remove unneeded check
Patch-mainline: v4.14-rc1
Git-commit: 660b1de13ce8a56cf8c1700774a962e36222c0f9
References: bsc#1050662 FATE#323951

Port validation is performed in ib_core, no need to duplicate it here.

Signed-off-by: Yuval Shaia <yuval.shaia@oracle.com>
Reviewed-by: Leon Romanovsky <leonro@mellanox.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/sw/rxe/rxe_av.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/infiniband/sw/rxe/rxe_av.c
+++ b/drivers/infiniband/sw/rxe/rxe_av.c
@@ -38,11 +38,6 @@ int rxe_av_chk_attr(struct rxe_dev *rxe,
 {
 	struct rxe_port *port;
 
-	if (rdma_ah_get_port_num(attr) != 1) {
-		pr_info("invalid port_num = %d\n", rdma_ah_get_port_num(attr));
-		return -EINVAL;
-	}
-
 	port = &rxe->port;
 
 	if (rdma_ah_get_ah_flags(attr) & IB_AH_GRH) {
